x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/QlClientIllegalArgumentException.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.ql;
+
+/**
+ * Exception thrown when unable to continue processing client request,
+ * in cases such as invalid query or failure to apply requested processing to given data.
+ */
+public class QlClientIllegalArgumentException extends QlClientException {
```

Comment:
bikeshed: we could also have a generic `InvalidQueryException` (and make `VerifierException` etc. inherit from it)

Reply:
Thanks.
I've updated it to a "middle" `InvalidArgumentException` which is what all updates turned out to refer to, as QL-specific client-side version of `QlIllegalArgumentException`. I think this makes more sense as a user-facing `"type"`.

Also, I think a specialisation isn't actually needed; i.e. no `EqlInvalidArgumentException` or `SqlInvalidArgumentException`, like in the case of `QlIllegalArgumentException`(whose specialisations aren't very clear to me -- left a couple of TODOs).